pkg/jira/search.go

```diff
@@ -32,6 +33,11 @@ func (c *Client) search(jql string, limit uint, ver string) (*SearchResult, erro
 		err error
 	)
 
+	// maxResults is a int32
+	// https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issue-search/#api-rest-api-3-search-get
+	if math.MaxInt32 < limit {
+		limit = math.MaxInt32
+	}
```

Comment:
I am not sure if this fix is necessary given that Jira API won't return the expected number of records even for numbers like 120 or 150 as the response payload is huge. The output we get when using the `limit` option is not reliable, and I think this feature needs to be redesigned to accept a range of items (eg: from 100, take 50).

Reply:
@ankitpokhrel point taken. Is there an issue for the problem where the returned payload < limit ? What recourse do we have if the Jira endpoint won't return _all_ the list items we request?

Reply:
@realtime-neil As per [the docs](https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issue-search/#api-rest-api-3-search-get)

```
The greatest number of items returned per page is achieved when requesting id or key only.
```

But if we only request for `id` or `key` to make the limit functionality work, we would need to send separate requests to get other details we are displaying in the TUI. And that's not feasible.

Reply:
point taken